tests/flow_counter/flow_counter_utils.py

```diff
@@ -0,0 +1,278 @@
+import logging
+import random
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.utilities import wait_until, check_skip_release
+
+logger = logging.getLogger(__name__)
+
+support_route_flow_counter = None
+skip_versions = ['201811', '201911', '202012', '202106', '202111']
+CAPABILITY_WAIT_TIME_IN_SEC = 180
+CAPABILITY_CHECK_INTERVAL_IN_SEC = 5
+
+
+class RouteFlowCounterTestContext:
+    """Allow caller to use "with" key words to run router flow counter test.
+    """
+    def __init__(self, dut, route_pattern_list, expected_stats, interval=1000):
+        """Init RouteFlowCounterTestContext
+
+        Args:
+            dut (object): DUT object
+            route_pattern_list (list): a list of route pattern, e.g. ['1.1.1.0/24', 'Vrf1|1.1.1.0/24', 'Vnet1|2.2.2.0/24']
+            expected_stats (dict): Expected result value. e.g. {'1.1.1.0/24': {'packets': '5', 'bytes': '4500'}}
+            interval (int, optional): Route flow counter query interval. Defaults to 1000.
+        """
+        self.dut = dut
+        self.route_pattern_list = route_pattern_list
+        self.expected_stats = expected_stats
+        self.interval = interval
+
+    def __enter__(self):
+        """Enable route flow counter and configure route pattern
+        """
+        if not is_route_flow_counter_supported(self.dut):
```

Comment:
I suggest to set this value as an instance variable when initializing the class rather than calling it multiple times

Reply:
will fix